pkg/rbac/sharedData.go

```diff
@@ -95,7 +166,8 @@ func sharedCacheValid(shared *SharedData) bool {
 
 	if (time.Now().Before(shared.csUpdatedAt.Add(time.Duration(config.Cfg.SharedCacheTTL) * time.Millisecond))) &&
 		(time.Now().Before(shared.nsUpdatedAt.Add(time.Duration(config.Cfg.SharedCacheTTL) * time.Millisecond))) &&
-		(time.Now().Before(shared.mcUpdatedAt.Add(time.Duration(config.Cfg.SharedCacheTTL) * time.Millisecond))) {
+		(time.Now().Before(shared.mcUpdatedAt.Add(time.Duration(config.Cfg.SharedCacheTTL) * time.Millisecond))) &&
+		(time.Now().Before(shared.propTypeTime.Add(time.Duration(config.Cfg.SharedCacheTTL) * time.Millisecond))) {
```

Comment:
Do we need to invalidate the cache based on property types updated time? I hope refreshing property types if a  property is not found in there would suffice?

Reply:
I like @SherinV 's approach of "appending to the cache" whenever we encounter a new property.  Currently we don't have any scenario where the property type changes after we have discovered it.
However, this could be a significant change. We can try on a new branch but don't discard the current implementation.

Reply:
This is a good point. Should we continue with current implementation and add that as an enhancement?